### DISCUSSION TITLE
Discuss the handling of unused function arguments

### DISCUSSION THREAD

**Discussion by socutes** [apache/rocketmq#6444] [General]
There are many functions in the current code that need to pass arguments, but the arguments are not used. The current discovery is mainly in the DefaultRequestProcessor in nameserver. It's a similar story elsewhere.

From a coding point of view, I don't think this makes sense. I understand that this parameter should be passed for future use. But it has not been used for a long time.

So, I think from a code optimization point of view, invalid parameter passing should be removed. Pass parameters as needed.

I'd like to hear from the community.

eg:
<img width="667" alt="image" src="https://user-images.githubusercontent.com/7479320/227072030-08eac5aa-3e1c-40f8-9e74-80d4ae7a7a1b.png">

<img width="920" alt="image" src="https://user-images.githubusercontent.com/7479320/227072006-57373776-e26f-4ce0-b2cc-b5eb9079db08.png">

**Comment by echooymxq** [comment]
Agree. Do it!

**Reply by socutes** [reply]
> Agree. Do it!

Yes, I will submit this change if the community thinks it can be done
